django/contrib/auth/forms.py

```diff
@@ -86,11 +86,15 @@ def clean_password2(self):
                 self.error_messages['password_mismatch'],
                 code='password_mismatch',
             )
+        self.instance.username = self.cleaned_data.get('username')
+        password_validation.validate_password(self.cleaned_data.get('password2'), self.instance)
         return password2
 
     def save(self, commit=True):
         user = super(UserCreationForm, self).save(commit=False)
-        user.set_password(self.cleaned_data["password1"])
+        password = self.cleaned_data['password1']
+        user.set_password(password)
+        password_validation.password_changed(password, self.instance)
```

Comment:
This positioning is copied from the other forms, but I think it's wrong here and there. If you use `form.save(commit=False)`, you'll get the `password_changed()` functions called twice. If you have a function that prevents password reuse, that'll be a problem. Also these lines appear to be untested.